File: app/src/main/java/com/boardgamegeek/sorter/AverageWeightAscendingSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;

public class AverageWeightAscendingSorter extends AverageWeightSorter {
	public AverageWeightAscendingSorter(@NonNull Context context) {
		super(context);
	}

	@StringRes
	@Override
	public int getTypeResource() {
		return R.string.collection_sort_type_average_weight_asc;
	}

	@StringRes
	@Override
	public int getSubDescriptionId() {
		return R.string.lightest;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/AverageWeightAscendingSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes

import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItemEntity

class AverageWeightAscendingSorter(context: Context) : AverageWeightSorter(context) {
    @StringRes
    public override val typeResId = R.string.collection_sort_type_average_weight_asc

    @StringRes
    public override val subDescriptionResId = R.string.lightest

    override fun sort(items: Iterable<CollectionItemEntity>) = items.sortedBy { it.averageWeight }
}
